src/Main.hs

```diff
@@ -5,15 +5,17 @@ module Main where
 import Hedgehog hiding (checkParallel)
 import Hedgehog.Internal.Property (GroupName(..), PropertyName(..))
 import System.Environment         (getArgs)
+import Data.Maybe (listToMaybe)
+import Data.Text (pack)
 
 import Echidna.Exec
 import Echidna.Solidity
 
 main :: IO ()
 main = getArgs >>= \case
   []  -> putStrLn "Please provide a solidity file to analyze"
-  f:_ -> do
-    (v,a,ts) <- loadSolidity f
+  filepath:args -> do
+    (v,a,ts) <- loadSolidity filepath $ fmap pack $ listToMaybe args
```

Comment:
nit, but I've tried to prefer `f . g $ h x` over `f $ g $ h x`

Reply:
curious why you prefer `f . g $ h x` over `f $ g $ h x`? i find the latter a bit easier to read (more uniform, 1 concept rather than 2) and readability important

Reply:
Consistency when moving from `$` to `<$>`

Reply:
how about `loadSolidity filepath $ pack <$> listToMaybe args`?

Reply:
that's fine with me